---
layout: distill
title: "Lecture 2: Bayesian Networks"
description: Overview of Bayesian Networks, their properties, and how they can  be helpful to model the joint probability distribution over a set of random variables. 
date: 2019-01-16

lecturers:
  - name: Eric Xing
    url: "https://www.cs.cmu.edu/~epxing/"

authors:
  - name: Cathy Su
    url: "https://ceesu.github.io"   
  - name: Angel C. Hernandez
    url: "https://github.com/ahernandez105"
  - name: Mark Cheung
    url: "#"
  - name: Xueqian Li
    url: "#"

editors:
  - name: Maruan Al-Shedivat  # editor's full name
    url: "https://www.cs.cmu.edu/~mshediva/"  # optional URL to the editor's homepage

abstract: >
---

# Readings
## The Bayesian network representation <d-cite key="Koller:2009:PGM:1795555"></d-cite>

### 3.1.1  Exploiting independence Properties
#### Standard vs compact parametrization of  independent random variables
Given random variables $$X_i$$ each representing the outcome of an independent coin toss, the standard parametrization of their joint distribution would be as follows:

$$ P(X_1, ... X_n) = P(X_1 = x_1, X_2=x_2, ..., X_n = x_n) = P(X_1 = x_1)P(X_2=x_2)...P(X_n=x_n) $$

Note there are 2 possibilities for each outcome $$x_i$$, this representation requires $$2^n$$ parameters. However, only $$2^n -1$$ will be independent parameters since the last probability is fully determined by the first $$2^n$$ (all probabilities sum to 1).

One simple way to reduce the number of parameters needed, would be to represent the probability that each coin toss lands heads as $$ \theta_1, ... \theta_n $$.  Then we have the following compact representation requiring only $$n$$ parameters:

$$ P(X_1, ... X_n) = \prod_i \theta_{X_i} $$

### 3.1.3 Naive Bayes
We can further express the joint distribution in terms of conditional probabilities. This is done in the Naive Bayes model.

 Instances of this model will include:
* class : some category $$ C \in \{c^1, ...c^k\} $$ which gives a prior on the value of each feature
* features: observed properties $$X_1, ... X_k$$ 

The model makes the strong  'naive' conditional independence assumption:

$$ P(x_i | C,  x_1, \dots, x_n) = P(x_i | C) $$

In words, _features are conditionally independent, given the class of the instance of this model_.  Thus the joint distribution of the Naive Bayes model factorizes as follows:

$$ P(C, X_1, ... X_n) = P(C) \prod_i P(X_i |C) $$

### 3.2.1 Bayesian networks

Bayesian networks use a graph whose nodes are the random variables in the domain, and whose edges represent conditional probability  statements. Unlike in the Naive Bayes model, Bayesian networks can also represent distributions that do not satisfy the naive conditional independence assumption. 

#### Definition 3.1: Bayesian Network (BN)
A bayesian network $$\mathcal{G}$$ is a directed acyclic graph, whose nodes represent random variables $$X_1, ...X_n $$. Let $$Pa_{X_i}^{\mathcal{G}}$$ denote the parents of $$X_i \in \mathcal{G}$$ and $$NonDescendants_{X_i}$$ denote the variables in the graph which are not descendants of $$X_i$$. Then $$\mathcal{G}$$ encodes the following set of conditional independence assumptions, also called the 'local independencies' and denoted by $$I_l(\mathcal{G})$$:

For each variable $$ X_i: (X_i \perp NonDescendants_{X_i} \vert Pa_{X_i}^{\mathcal{G}} ) $$

### 3.2.3 Graphs and Distributions

In this section it is shown that the distribution $$P$$ satisfies the local independencies associated with  $$\mathcal{G} \iff P$$ is representable as a set of conditional probability distributions (CPDs) associated with $$\mathcal{G}$$. 

#### Definition 3.2-3.3: I-Map
Let $$P$$ be a distribution over $$\mathcal{X}$$. We define $$I(P)$$ to be the set of independence assertions of the form $$ X \perp Y \vert Z $$ that hold in $$P$$.

Let $$\mathcal{K}$$ be any graph object associated with a set of independencies $$I(\mathcal{K})$$. We say $$\mathcal{K}$$ is an I-map for $$I$$ if $$I(\mathcal{K}) \subseteq I$$.

**Note this means that $$\mathcal{G}$$ is an I-map for $$P $$ if $$\mathcal{G}$$ is an I-map for $$I(P)$$.**

#### I-Map to factorization

In this section  it is proven (see text) that the conditional independence assumptions implied by the BN structure $$\mathcal{G}$$ allow us to factorize the distribution $$P$$ for which $$\mathcal{G}$$ is an I-map into conditional probability distributions. 

##### Definition 3.4 Factorization
Let $$P$$ be a distribution and $$\mathcal{G}$$ be a BN graph over random variables $$X_1, ...X_n $$. We say that $$P$$ _factorizes_ according to $$\mathcal{G}$$ if $$P$$ can be expressed as a product:

$$ P(X_1, ... X_n) = \prod_i P( X_i | Pa_{X_i}^{\mathcal{G}}) $$

##### Reduction in number of parameters
In a distribution over $$n$$ binary variables, specifying the joint distribution will require $$2^n - 1$$ independent parameters (as stated in 3.1.1). However if the distribution factorizes according to  $$\mathcal{G}$$ where each node has at most $$k$$ parents, then the number of independent parameters is $$ < 2^k$$. **Since $$k$$ is usually small, this represents an exponential reduction in number of parameters.**

#### Factorization to I-map
The converse also holds as given by the following theorem:
##### Theorem 3.2
Let  $$\mathcal{G}$$ be a BN graph over random variables $$X $$ and $$P$$ be a distribution over the same space. If $$P$$ factorizes according to $$\mathcal{G}$$ then $$\mathcal{G}$$ is an I-map for  $$P$$.

### Box 3.C Knowledge engineering

 Building a BN in the real world requires many steps including:

* _picking variables precisely_: we need some variables that can be observed, and their domain should be specified. Sometimes introducing hidden variables will help to render the observed variables conditionally independent.
* _picking a structure consistent with the causal order_: choose enough variables to approximate the causal relationships. 
* _picking probability estimates for events in the network_: data collection may be difficult but small errors have little effect (though the network is sensitive to events assigned a zero probability, and to relative size of probabilities of events)

Done correctly, the model will be useful as well as not too complex to use (see Box 3.D).

### 3.3.1 D-separation

**Objective:** determine which independencies hold for every distribution $$P$$ which factorizes over $$G$$.

#### Definition 2.16 Trail

We say that $$X_1, ... X_k$$ form a trail in the graph $$\mathcal{K} = (\mathcal{X}, \mathcal{E}) $$ if for every $$i = 1, ... , k-1$$ we have that $$ X_i \rightleftharpoons X_{i+1}$$. 

#### Types of active two edge trails 

By examining 2-edge connections between nodes $$X,Y,Z$$ four types of active trails, i.e. trails along which influence can flow from $$X$$ to $$Z$$, are apparent:

1. Causal trail $$X \rightarrow Z \rightarrow Y$$ : active iff Z is not observed. 
2. Evidential trail $$X \leftarrow Z \leftarrow Y$$ : active iff Z is not observed.
3. Common cause $$X \leftarrow Z \rightarrow Y$$ : active iff Z is not observed.
4. Common effect $$X \rightarrow Z \leftarrow Y$$ : active iff Z or one of its descendants is observed.

For influence to flow through a longer trail $$ X_1 \rightleftharpoons ... \rightleftharpoons X_n  $$, every two-edge trail within must allow influence flow. This is summarized as follows:

#### Definition 3.6 active trail
Let $$\mathcal{G}$$ be a BN structure and. $$ X_1 \rightleftharpoons  ... \rightleftharpoons X_n  $$ be a trail in $$\mathcal{G}$$. Let $$Z$$ be a subset of observed variables. Then the trail $$ X_1 \rightleftharpoons  ... \rightleftharpoons  X_n$$ is active given $$Z  $$ if:

* Whenever we have a v-structure $$ X_{I-1}  \to X_i \leftarrow X_{I+1}  $$ then $$X_i$$ or one of its descendants are in $$Z$$
* No other node along the trail is in $$Z$$

Further, for directed graphs that may have more than one trail between nodes "directed separation" (d-separation) gives a notion of separation between nodes.

#### Definition 3.7 D-separation 
Let $$X,Y,Z$$ be three sets of nodes in $$\mathcal{G}$$. We say that *$$X, Y$$* are d-separated given $$Z$$ (denoted $$ \text{d-sep}_{\mathcal{G}} (X; Y \vert Z) $$ ) if there is no active trail between any node X $$ \in X $$ and Y $$ \in Y $$ given $$Z$$. 

### 3.3.2 Soundness and Completeness
As a method d-separation has the following properties (proof in text):

* _Soundness_: $$ X, Y $$ are d-separated given $$ Z \implies X \perp Y \vert Z $$ i.e. independence determined by d-separation is satisfied by the underlying distribution. 
* _Completeness_:  If $$ X \perp Y \vert Z $$ then they are d-separated, i.e. d-separation detects all possible independencies.

***

# In class notes

**Motivation:** representing joint distributions over some random variables is computationally expensive, so we need methodologies to represent joint distributions compactly.

## Two types of Graphical Models <d-cite key="Blei:2015:BasicsPGM"></d-cite><d-cite key="Jordan:2003:PGM"></d-cite>
### Directed Graphs (Bayesian Networks)
An acyclic graph, $$\mathcal{G}$$, is made up of a set of nodes, $$\mathcal{V}$$, and a set 
of directed edges, $$\mathcal{E}$$, where edges represent a causality relationship between nodes. Nodes in the graph represent a set of random variables, $$\{x_1,...,x_N\}$$, where there is a one-to-one map between nodes and random variables. 

<img src="{{ '/assets/img/notes/lecture-02/directed-graph.png' | relative_url }}" height ="270" width = "480"/>

The joint probability of the above directed graph can be written as follows:

$$(\{X_1,...,X_8\}) = P(X_1)P(X_2)P(X_3|X_1)P(X_4|X_2)P(X_5|X_2)P(X_6|X_3,X_4)P(X_7|X_6)P(X_8|X_5,X_6)$$

### Undirected Graphs (Markov Random Fields) <d-cite key="Jordan:2003:PGM"></d-cite>
An undirected graph contains nodes that are connected via non-directional edges.

<img src="{{ '/assets/img/notes/lecture-02/undirected-graph.png' | relative_url }}" height ="270" width = "480" />

The joint probability of the above undirected graph can be written as followed:

$$P(\{X_1,...,X_8\})= \frac{1}{Z}exp\bigg(E(X_1)+E(X_2)+E(X_3,X_1)+E(X_4,X_2)+E(X_5,X_2)+E(X_6,X_3,X_4)+E(X_7,X_6)+E(X_8,X_5,X_6) \bigg)$$

## Notation

* Variable: capitalized english letter, with subscripts to represent dimensions (i, j, k) and superscripts to represent index e.g. $$V_{i, j}^j$$ 
* Values of variables: a lowercase letter means it is an 'observed value' of some random variable e.g. $$v_{i, j}^j$$
* Random variable: a variable with stochasticity, changing across different observations
* Random vector: capitalized and bold letter, with random vars as entries (of dimension 1xn)
* Random matrix: capitalized and bold letter, with random vars as entries (of dimension nxm)
* Parameters: greek characters. Can be considered random variables

## The Dishonest Casino

Let $$\textbf{x}$$ be a sequence of random variables, $$x_1,...,x_T$$, where $$x_t$$ is the outcome 
of a casino's die roll, $$ x \in (1,2,3,4,5,6) $$. Also, let $$\textbf{y}$$ be a parse of random
variables, $$y_1,...,y_T$$, where $$y_t$$ is the outcome of whether the die was fair or bias,$$ y \in (0,1)$$. The bias die follows the following distributions:

p(x=1)|p(x=2)|p(x=3)|p(x=4)|p(x=5)|p(x=6)|
-----|-------|------|------|------|------|
0.1|0.1|0.1|0.1|0.1|0.5|

Some questions we might want to ask are:

* **Evaluation:** How likely is the sequence, given our model of casino?
* **Decoding**: What portion of the sequence was generated
with the fair die, and what portion with the loaded die?
* **Learning:** How "loaded" is the loaded die? How "fair" is the fair die? How often does the casino player change from fair die to loaded die, and back?

One way we could model this casino problem is as a Hidden Markov Model where $$x_t$$ are our
observed variables and $$y_t$$ are our hidden variables.

<img src="{{ '/assets/img/notes/lecture-02/hmm.png' | relative_url }}" height ="270" width = "480"/>

The hidden variables all share the _Markov property_ that the the past is conditionally 
independent of the future given the present:

$$y_{t-1} \, \bot \, \{y_{t+1},...,y_{T}\} \, | \, y_t$$

This property is also explicitly highlighted in the topology of the graph. 

Furthermore, we can find how likely of the parse, given our HMM sequence, as followed:

$$p(\textbf{x,y}) = p(y_1)\prod\limits_{t=1}^T p(x_t\,|\,y_t)\prod\limits_{t=2}^{T} p(y_t\,|\,y_{t-1})$$

$$p(\textbf{x,y}) = p(y_1)p(x_1|y_1)p(y2|y_1)p(x_2|y_2)...p(y_T|y_{T-1})p(x_T|y_T)$$

$$p(\textbf{x,y}) = p(y_1)p(y_2|y_1)...p(y_T|y_{T-1})\times p(x_1|y_1)p(x_2|y_2)...p(x_T|y_T)$$

$$p(\textbf{x,y}) = p(y_1,...,y_T)p(x_1,...,x_T|y_1,...,y_T)$$

The marginal and posterior distributions can be computed as followed:

$$Marginal: p(\textbf{x}) = \sum\limits_{y_{1}}\cdots\sum\limits_{y_T} p(\textbf{x,y})$$

$$Posterior: p(\textbf{y}\,|\,\textbf{x}) = \frac{p(\textbf{x,y})}{p(\textbf{x})}$$

## Bayesian Network

* A BN is a directed graph whose nodes represent the random variables and whose edges represent 
directed influence of one variable on the another.
* It is a data structure that provides the skeleton for representing **a joint distribution** compactly in a **factorized** way.
* It offers a compact representation for **a set of conditional independence assumptions** about a distribution.
* We can view the graph as encoding a **generative sampling process** executed by nature, where 
* the value for each variable is selected by the nature using a distribution that depends only on its 
parents. In other words, each variable is a stochastic function of its parents.

## Bayesian Network: Factorization Theorem
We define  $$x_{\pi_i}$$ and $$x_{\iota_i}$$ as the set of parent nodes and non-ancestor nodes 
for the $$i$$ node, respectively. The topology of a directed graph asserts the set of conditional independence statements:

$$ \{ x_i \, \bot \, x_{\iota_i} \, | \, x_{\pi_i} \} $$

<img src="{{ '/assets/img/notes/lecture-02/directed-graph.png' | relative_url }}" height ="270" width = "480"/>

As a result, the joint probability of the above directed graph can be written as follows:

$$P(\{X_1,..,X_8 \}) = \prod\limits_{i =1}^8 P(x_i \, | \, x_{\pi_i})$$

$$P(\{X_1,...,X_8\}) = P(X_1)P(X_2)P(X_3|X_1)P(X_4|X_2)P(X_5|X_2)P(X_6|X_3,X_4)P(X_7|X_6)P(X_8|X_5,X_6)$$

## Specification of a Directed Graphical Model

There are two components to any GM:

* qualitative (topology)
* quantitative (numbers associated with each conditional distributions)

### Sources of Qualitative Specifications

Where do our assumptions come from?
* Prior knowledge of causal relationship
* Prior knowledge of modular relationship
* Assessment from expert
* Learning from data
* We simply like a certain architecture (e.g. a layered graph)
* ...

## Local Structures & Independencies

* Common parent
  * Fixing B decouples A and C
  * "given the level of gene B, the levels of A and C are independent"

<img src="{{ '/assets/img/notes/lecture-02/common_parent.png' | relative_url }}" />

$$ A \perp C |B => P(A,C|B)=P(A|B)P(C|B) $$

* Cascade
  * Knowing B decouples A and C
  * "given the level of gene B, the level gene A provides no extra prediction value for the level of gene C"

<img src="{{ '/assets/img/notes/lecture-02/cascade.png' | relative_url }}" />

* V-structure
  * Knowing C couples A and B because A can "explain away" B w.r.t. C
  * "If A correlates to C, then chance for B to also correlate to B will decrease"
  
<img src="{{ '/assets/img/notes/lecture-02/V-structure.png' | relative_url }}" />

#### In class example of v-structure

My clock running late (event A) and a traffic jam (event B) which are independent events could both cause me to be late for class (event C). However, my presence couples the two events. Observe that if my clock was running late it could 'explain away' my lateness due to the traffic jam.

i.e. ($$ A $$ and $$ B $$ become coupled if $$ C $$ is known because they jointly cause the 3rd event): $$  P(A,B)=P(A)P(B) $$ but $$ P (A,B \vert C) $$ cannot be decoupled into two conditionals
  
* The language is compact, the concepts are rich!

### A simple justification

Prove: $$ A \perp C \vert B => P(A,C \vert B)=P(A \vert B)P(C \vert B) $$

From the motif: $$P(A,B,C)=P(B)P(A \vert B)P(C \vert B)$$

$$P(A,C \vert B)=P(A,B,C)/P(B)$$

Plugging in the above,  $$P(A,C \vert B)=P(B)P(A \vert B)P(C \vert B)/P(B)=P(A \vert B)P(C \vert B)$$

## I-maps

* Defn: Let $$P$$ be a distribution over $$X$$. We define $$I(P)$$ to be the set of independence assertions of the form $$(X \perp Y \vert Z)$$ that hold in $$P$$ (however how we set the parameter-values).

* Defn: Let $$K$$ be any graph object associated with a set of independencies $$I(K)$$. We say that $$K$$ is an I-map for a set of independencies $$I$$, if $$I(K) \subseteq I$$.

* We now say that $$G$$ is an I-map for $$P$$ if $$G$$ is an I-map for $$I(P)$$, where we use $$I(G)$$ as the set of independencies associated.

## Facts about I-map

* For $$G$$ to be an I-map of $$P$$, it is necessary that $$G$$ does not mislead us regarding independencies in $$P$$:

  any independence that $$G$$ asserts must also hold in $$P$$. Conversely, $$P$$ may have additional independencies that are not reflected in $$G$$
 
### In class example
<img width = "480" src="{{ '/assets/img/notes/lecture-02/I-maps.png' | relative_url }}" />

$$X$$ and $$Y$$ are independent only in Graph 1.

Below we have two example marginal distributions. Find the I-maps: 

* $$ P_1: I(P_1) = {X \perp Y} $$ (from inspection i.e., $$ P(X,Y)=P(X)P(Y), 0.48=0.6* 0.8 $$)

<img src="{{ '/assets/img/notes/lecture-02/I-Map_p1.png' | relative_url }}" />

Solution: Graph 1.

* $$P_2$$: $$I(P_2) = \emptyset $$

<img src="{{ '/assets/img/notes/lecture-02/I-Map_p2.png' | relative_url }}" />

Solution: Both Graph 2 and Graph 3, since they assume no independences. Therefore, they are equivalent in terms of independent sets. Graph 1 has an independent set. Therefore it is not an I-map of $$P_2$$.

## What is $$I(G)$$

### local Markov assumptions of BN

A *Bayesian network structure* $$G$$ is a directed acyclic graph whose nodes represent random variables $$X_1, . . . ,X_n$$

### local Markov assumptions

#### Definition

Let $$Pa_{X_i}$$ denote the parents of $$X_i$$ in $$G$$, and $$ NonDescendants_{X_i} $$ denote the variables in the graph that are not descendants of $$X_i$$. Then $$G$$ encodes the following set of local conditional independence assumptions $$I_{\ell}(G)$$:

$$ I_{\ell} (G): \left\{ X_i \perp NonDescendants_{X_i} \vert Pa_{X_i}: \forall i \right\} $$

In other words, each node $$X_i$$ is independent of its nondescendants given its parents.

## D-separation criterion for Bayesian networks

#### Definition 
Variables x and y are *D-separated* (conditionally independent) given z if they are separated in the *moralized* ancestral graph.  (D for Directed edges)

* Example: $$ X \perp Y \vert Z$$, then we say $$Z$$ D-separates $$X$$ and $$Y$$.

<img src="{{ '/assets/img/notes/lecture-02/graph_separation_criterion_example.png' | relative_url }}" width="500" />

* Ancestral graph (focusing only the nodes of interest): keeping only the nodes themselves + ancestors of the nodes in question.

* Moral ancestral: remove the connectivity between nodes, but connect parents to nodes that are shared by a single child.

If there is a way to travel from one node to another node using any path (not through the given), then these two nodes are not conditionally independent (example is not conditionally independent).


## What is in $$I(G)$$

Global Markov properties of Bayesian networks
* X is *d-separated* (directed-separated) from Z given Y if we can't send a ball from any node in X to any node in Z using the *"Bayes-ball"* algorithm illustrated bellow (and plus some boundary conditions):

<img src="{{ '/assets/img/notes/lecture-02/d_separated.png' | relative_url }}" />

* Defn: $$I(G) =$$ all independence properties that correspond to d-separation:

$$ I(G): \left\{ X_i \perp Z \vert Y: dsep_G ( X ; Z \vert Y) \right\} $$

* D-separation is sound and complete (see 'Soundness and Completeness' section below)

## I(G) Example 

<img src="{{ '/assets/img/notes/lecture-02/IG_Exp.png' | relative_url }}" />

In this graph there are two types of active trail structures (see section 3.3.1 above from the reading for definition):

* common cause: $$x_3 \leftarrow x_1 \rightarrow x_4$$. This trail is active if we don't condition on $$x_1$$.
* common effect: $$x_2 \rightarrow x_3 \leftarrow x_1$$. This trail is active if we condition on $$x_3$$.

To find the independencies, consider all trails with length greater than 2 (since a node cannot be independent of its parent). 

### Trails of length 3:
* $$x_2 \rightleftharpoons x_3 \rightleftharpoons x_1$$: due to 'common effect' structure, this trail is blocked as long as we do not condition on $$x_3$$. Therefore we get $$(x_2 \perp \{x_1\}), (x_2 \perp \{x_1\} \vert x_4)$$
* $$x_3 \rightleftharpoons  x_1 \rightleftharpoons x_4$$: due to 'common cause' structure, this trail is blocked if we condition on $$x_1$$. Therefore we get $$(x_3 \perp \{x_4\} \vert x_1), (x_3 \perp \{x_4\} \vert \{x_1, x_2\})$$

### Trails of length 4 (only $$x_2 \rightleftharpoons  x_3 \rightleftharpoons x_1 \rightleftharpoons x_4$$):
* due to 'common effect' structure $$x_2 \rightarrow x_3 \leftarrow x_1$$, this trail is blocked as long as we do not condition on $$x_3$$. Therefore we get $$(x_2 \perp \{x_4\}),  (x_2 \perp \{x_4\} \vert x_1)$$.
* due to 'common cause' structure $$x_3 \leftarrow x_1 \rightarrow x_4$$, this trail is blocked if we condition on $$x_1$$. Therefore we get $$(x_2 \perp \{x_4\} \vert x_1), (x_2 \perp \{x_4\} \vert \{x_1, x_3\})$$

### Full $$I(\mathcal{G})$$ 
Putting the above together we have the following independencies.

$$\begin{aligned}
I(\mathcal{G}) 
&= \{(x_2 \perp \{x_1\}), (x_2 \perp \{x_1\} \vert x_4), \\
&(x_3 \perp \{x_4\} \vert x_1), (x_3 \perp \{x_4\} \vert \{x_1, x_2\}), \\
&(x_2 \perp \{x_4\}),  (x_2 \perp \{x_4\} \vert x_1),  (x_2 \perp \{x_4\} \vert \{x_1, x_3\}) 
\end{aligned}$$

## The Equivalence Theorem 

For a graph $$G$$, Let $$D_1$$ denotes the family of all distributions that satisfy $$I(G)$$. Let $$D_2$$ denotes the family of all distributions that factor according to $$G$$. Then $$D_1 \equiv D_2$$, which can be expressed as 

$$ P(X)=\prod_{i=1:d}P(X_i \vert X_{\pi_i}) $$.

This means separation properties in the graph imply independence properties about the associated variables.

## Conditional Probability Tables (CPTs)

This is an example with discrete probabilities.

<img src="{{ '/assets/img/notes/lecture-02/CP_1.png' | relative_url }}" />


## Conditional Probability Density Func. (CPDs) 

The probabilities can also be a continuous function, e.g. the Gaussian distribution.

<img src="{{ '/assets/img/notes/lecture-02/CP_2.png' | relative_url }}" />

## Summary of BN

* Conditional independencies imply factorization.

* Factorization according to $$G$$ implies the associated conditional independencies.

## Soundness and Completeness of D-separation 

* Soundness:

    Theorem: If a distribution $$P$$ factorizes according to $$G$$, then $$I(G)\subseteq I(P)$$ 
* "Completeness":

    "Claim": For any distribution $$P$$ that factorizes over $$G$$, if $$(X \perp Y \vert Z) \in I(P)$$, then $$ \text{d-sep}_{\mathcal{G}} (X; Y \vert Z) $$ )

  * Actually, the "Completeness" is not holding true all the time. "Even if a distribution factorizes over G, it can still contain additional independencies that are not reflected in the structure."
  
  * Example:

    Given graph $$ A->B $$, and a conditional probability table,
    <img src="{{ '/assets/img/notes/lecture-02/CPT.png' | relative_url }}" />
    you can hard code in a subtle way to capture some additional independencies in some specific distributions.
  
  * **Thm**:

    Let $$G$$ be a $$BN$$ graph. If $$X$$ and $$Y$$ are not d-separated given $$Z$$ in $$G$$, then $$X$$ and $$Y$$ are dependent in  **some** distribution $$P$$ that factorizes over $$G$$.

* **Theorem**:
  For **almost all** distributions $$P$$ that factorize over $$G$$, i.e., for all distributions except for a set of "measure zero" in the space of CPD parameterizations, we have that $$I(P)=I(G)$$.
  
